src/simsopt/_core/derivative.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+import numbers
+import collections
+from .graph_optimizable import Optimizable
+
+
```

Comment:
I don't see any use for this class outside the derivative for now. If such a use arises, I'm happy to move the Optimizable import inside the `__missing__` function and move that class into `_core.util`. For now I think it makes sense to keep it where it is, since it's only used by the derivative.